src/workers/filesystems/infrastructure/SyncLocalFileSystem.ts

```diff
@@ -0,0 +1,341 @@
+  }
+
+  async function getLocalMeta(pathname: string): Promise<LocalItemMetaData> {
+    const stat = await fs.stat(pathname);
+    return LocalItemMetaData.from({
+      modtime: Math.trunc(stat.mtimeMs / 1000),
+      size: stat.size,
+      ino: stat.ino,
+      dev: stat.dev,
+      isFolder: stat.isDirectory(),
+    });
+  }
+
+  return {
+    kind: 'LOCAL',
+    async getCurrentListing() {
+      const ig = ignore().add(ignoredFiles);
+
+      const list = (
+        await glob('**', {
+          filesOnly: false,
+          absolute: true,
+          dot: true,
+          cwd: localPath,
+        })
+      ).filter((itemPath) => {
+        const relativeItemName = path.relative(localPath, itemPath);
+
+        if (ig.ignores(relativeItemName)) {
+          return false;
+        }
+
+        const isValid = fileNameIsValid(relativeItemName);
+        if (!isValid) {
+          Logger.warn(
+            `${this.kind} file with name ${relativeItemName} will be ignored due an invalid name`
+          );
+          return false;
+        }
+
+        return true;
+      });
+
+      const listing: LocalListing = {};
+      const readingMetaErrors: ReadingMetaErrorEntry[] = [];
+
+      for (const itemName of list) {
+        const relativeName = getListingPath(itemName);
+        try {
+          const metadata = await getLocalMeta(itemName);
+
+          if (!metadata.isEmpty()) listing[relativeName] = metadata;
+          else {
+            const emptyFileError = {
+              message: 'Internxt does not support empty files',
+              code: '',
+              stack: '',
+            };
+            readingMetaErrors.push({
+              name: relativeName,
+              errorName: 'EMPTY_FILE',
+              errorDetails: createErrorDetails(
+                emptyFileError,
+                'Reading metadata of file',
+                `Size of file is ${metadata.size}`
+              ),
+            });
+          }
+        } catch (e) {
+          const err = e as { code?: string };
+
+          readingMetaErrors.push({
+            name: relativeName,
+            errorName: err.code === 'EPERM' ? 'NO_PERMISSION' : 'UNKNOWN',
+            errorDetails: createErrorDetails(
+              err,
+              'Reading metadata of file',
+              `File name is ${itemName}`
+            ),
+          });
+        }
+      }
+      return { listing, readingMetaErrors };
+    },
+
+    async deleteFile(name: string) {
+      const actualPath = getActualPath(name);
+      try {
+        await fs.unlink(actualPath);
+      } catch (e) {
+        const err = e as { code?: string };
+
+        if (err.code !== 'ENOENT')
+          throw new ProcessError(
+            'UNKNOWN',
+            createErrorDetails(err, 'Deleting a file locally', `Name: ${name}`)
+          );
+      }
+    },
+
+    pullFile(name: string, source: Source) {
+      return new Promise(async (resolve, reject) => {
```

Comment:
do not mix async and Promise, please use try-catch plus async-await

Reply:
If you need to catch the error to convert it, do this inside the catch:

`throw new ProcessError(...`